crates/consensus/src/block/header.rs

```diff
@@ -622,24 +679,54 @@ pub trait BlockHeader {
     /// Retrieves the requests hash of the block, if available
     fn requests_hash(&self) -> Option<B256>;
 
+    /// Retrieves the target blob count of the block, if available
+    fn target_blobs_per_block(&self) -> Option<u64>;
+
     /// Retrieves the block's extra data field
     fn extra_data(&self) -> &Bytes;
 
     /// Calculate excess blob gas for the next block according to the EIP-4844
     /// spec.
     ///
+    /// If [`BlockHeader::target_blobs_per_block`] is [`Some`], uses EIP-7742 formula for
+    /// calculating the excess blob gas, otherwise uses EIP-4844 formula.
+    ///
+    /// Note: this function will return incorrect (unnormalized, lower) value at EIP-7742 activation
+    /// block. If this is undesired, consider using [`eip7742::calc_excess_blob_gas_at_transition`].
```

Comment:
I think this should be OK asI'd expect all consumers of this to not know whether the next block activates 7742 if it's not an EL

Reply:
yep